tools/codemod/src/utils/TransformUtils.ts

```diff
@@ -0,0 +1,128 @@
+import { ObjectExpression, ObjectProperty, JSCodeshift, ASTPath, JSXElement, JSXIdentifier, JSXExpressionContainer, Identifier, ArrayExpression, ArrowFunctionExpression, AssignmentExpression, AwaitExpression, BigIntLiteral, BinaryExpression, BindExpression, BooleanLiteral, CallExpression, ChainExpression, ClassExpression, ComprehensionExpression, ConditionalExpression, DirectiveLiteral, DoExpression, FunctionExpression, GeneratorExpression, Import, ImportExpression, JSXFragment, JSXMemberExpression, JSXText, Literal, LogicalExpression, MemberExpression, MetaProperty, NewExpression, NullLiteral, NumericLiteral, OptionalCallExpression, OptionalMemberExpression, ParenthesizedExpression, PrivateName, RegExpLiteral, SequenceExpression, StringLiteral, Super, TaggedTemplateExpression, TemplateLiteral, ThisExpression, TSAsExpression, TSNonNullExpression, TSTypeAssertion, TSTypeParameter, TypeCastExpression, UnaryExpression, UpdateExpression, YieldExpression, SpreadProperty, ASTNode, ArrayPattern, AssignmentPattern, ObjectPattern, PropertyPattern, RestElement, SpreadElementPattern, SpreadPropertyPattern, TSParameterProperty, Property } from "jscodeshift";
+
+export type ExpressionKind = Identifier | FunctionExpression | ThisExpression | ArrayExpression | ObjectExpression | Literal | SequenceExpression | UnaryExpression | BinaryExpression | AssignmentExpression | MemberExpression | UpdateExpression | LogicalExpression | ConditionalExpression | NewExpression | CallExpression | ArrowFunctionExpression | YieldExpression | GeneratorExpression | ComprehensionExpression | ClassExpression | Super | TaggedTemplateExpression | TemplateLiteral | MetaProperty | AwaitExpression | ImportExpression | ChainExpression | OptionalCallExpression | OptionalMemberExpression | JSXIdentifier | JSXExpressionContainer | JSXElement | JSXFragment | JSXMemberExpression | JSXText | PrivateName | TypeCastExpression | DoExpression | BindExpression | ParenthesizedExpression | DirectiveLiteral | StringLiteral | NumericLiteral | BigIntLiteral | NullLiteral | BooleanLiteral | RegExpLiteral | Import | TSAsExpression | TSNonNullExpression | TSTypeParameter | TSTypeAssertion;
```

Comment:
Code smell. Can we figure out a better way w/o redefining this type?

Reply:
As we discussed earlier, since `jscodeshift` uses `ast-types` dependency and does not expose `Kinds`, the only ways to redefine these that I can think of are: 1) use `ast-types` dependency ourselves 2) use lookup types on some loosely related `ast-type` interface that `jscodeshift` exposes (but imo that would smell even worse) 3) use `Expression` interface instead of `ExpressionKind`, but that would require casting it every time we use some `jscodeshift` builder method that requires parameter of type `ExpressionKind`

Reply:
We've also discussed, that the need for these types indicates misuse of the indended `jscodeshift` API. One solution would be to use https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html, but ideally I'd like to get rid of these types altogether.